core/src/main/java/hudson/model/listeners/ItemListener.java

```diff
@@ -205,6 +218,19 @@ public static void fireOnUpdated(final Item item) {
         Listeners.notify(ItemListener.class, false, l -> l.onUpdated(item));
     }
 
+    @Restricted(NoExternalUse.class)
+    public static void checkBeforeDelete(Item item) throws Failure, InterruptedException {
+        for (ItemListener l : all()) {
```

Comment:
Like `checkBeforeCopy`, does _not_ use `Listeners.notify`, and runs with the same authn as `delete`.